NExtends/Expressions/ExpressionEqualityComparer.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NExtends.Expressions
+{
+	/// <summary>
+	/// source : http://stackoverflow.com/questions/283537/most-efficient-way-to-test-equality-of-lambda-expressions
+	/// </summary>
+	public static class ExpressionEqualityComparer
+	{
+		public static bool Eq(LambdaExpression x, LambdaExpression y)
+		{
+			return ExpressionsEqual(x, y, null, null);
+		}
+
+		public static bool Eq<TSource1, TSource2, TValue>(Expression<Func<TSource1, TSource2, TValue>> x, Expression<Func<TSource1, TSource2, TValue>> y)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of generic parameters in the 'ExpressionEqualityComparer.Eq' method to no more than the 2 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2436)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.Expression' instead of 'System.Linq.Expressions.Expression<System.Func<TSource1, TSource2, TValue>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.Expression' instead of 'System.Linq.Expressions.Expression<System.Func<TSource1, TSource2, TValue>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)